### PR TITLE
Restrict LAPACKFullMatrix::print_formatted to matrix states

### PR DESCRIPTION
**Pull request by masterleinad**
Currently, it is possible to call `LAPACKFullMatrix::print_formatted` also in case, e.g. `compute_svd()`, `compute_inverse_svd()` or `compute_lu_factorization()` is called. It is not immediately obvious that the stored (and thus printed) indices don't resemble the matrix or its inverse anymore.
Although this change is strictly speaking not backwards compatible, I doubt that anyone can use the output if the state is neither `matrix` nor `inverse_matrix` as the stored entries are just internal data otherwise.

### CODE REVIEW

File: include/deal.II/lac/lapack_support.h [hunk 1]
```diff
@@ -26,7 +26,7 @@ DEAL_II_NAMESPACE_OPEN
 namespace LAPACKSupport
 {
   /**
-   * Most LAPACK functions change the contents of the matrix applied to to
```
Comment (drwells) [inline include/deal.II/lac/lapack_support.h]: :)

Comment (drwells) [inline include/deal.II/lac/lapack_support.h]: I read this a few more times and it appears the original is actually correct, just a little difficult to read: the first clause is "Most LAPACK functions change the contents of the matrix applied to" and the second is "to something which is not a matrix anymore".

Comment (masterleinad) [inline include/deal.II/lac/lapack_support.h]: Ah, yes, now I see. This just look like a typical typo. Maybe reformatting ist still not a bad idea.

Comment (bangerth) [inline include/deal.II/lac/lapack_support.h]: How about rewording this as follows:
```
Most of the LAPACK functions one can apply to a matrix (e.g., by calling the member functions of this class) change its content in some ways. For example, they may invert the matrix, or may replace it by a matrix whose columns represent the eigenvectors of the original content of the matrix.

The elements of this enumeration are therefore used to track what is currently being stored by this object.
```

**Comment by bangerth** [comment]
OK to merge if you would change the comment to something more readable.

File: include/deal.II/lac/lapack_support.h (hunk 1 above)
Comment (masterleinad) [inline include/deal.II/lac/lapack_support.h]: Sure, here you go.

**Comment by bangerth** [comment]
There is an error in the CI output, but I cannot imagine it being a result of this patch. (It is about a file `boost/config.hpp` not being found when included from another boost file.) Thus, I'll merge and we'll just see what happens.
